nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -421,10 +424,13 @@ def setup_training_data(self, training_data_config=None):
 
     def setup_validation_data(self, validation_data_config=None):
         if self.cfg.data.get('validation_ds', None):
+            max_seq_length = self.frozen_model.cfg.encoder_seq_length
```

Comment:
What's the benefit of asking users to specify max_seq_length instead of using the default `self.frozen_model.cfg.encoder_seq_length`. Is there a situation in which they will be different? I can foresee how this might be a problem if `max_seq_length > self.frozen_model.cfg.encoder_seq_length`, so would prefer to leave it to default `self.frozen_model.cfg.encoder_seq_length` unless there's a clear advantage that I'm missing.

Reply:
it makes training inefficient. consider if 99.99999% of training examples are less than 50 tokens. But 1 example is 4000. We have to make mbs=1 just to fit this one long example. If we remove it we could probably make mbs=64 or 128.

Reply:
Ideallly the validation tool should suggest a good default here.

Reply:
I think this makes sense - how would you like the validation tool to suggest a good default? Maybe the length at which will preserve say 95/99% of samples?

Reply:
Current the validation tool drops samples > 10000 chars (which is around 2048 tokens) at line https://github.com/NVIDIA/NeMo/blob/main/tools/customization_dataset_preparation/customization_dataset_preparation.py#L268

Reply:
95 is too aggressive, 99% seems ok to me. But this is a separate discussion.